libraries/ESP8266WiFiMesh/src/CompatibilityLayer.cpp

```diff
@@ -0,0 +1,162 @@
+* NOTE!
+*
+* All method signatures in this file are deprecated and will be removed in core version 2.5.0.
+* If you are still using these methods, please consider migrating to the new API shown in 
+* the ESP8266WiFiMesh.h source file.
+*
+* TODO: delete this file.
+********************************************************************************************/
+
+
+
+
+
+#include <Arduino.h>
+#include <ESP8266WiFi.h>
+#include <WiFiClient.h> 
+#include <WiFiServer.h>
+
+#include "ESP8266WiFiMesh.h"
+
+#define SSID_PREFIX      		"Mesh_Node"
+#define SERVER_IP_ADDR			"192.168.4.1"
+#define SERVER_PORT				4011
+
+// DEPRECATED!
+ESP8266WiFiMesh::ESP8266WiFiMesh(uint32_t chip_id, std::function<String(String)> handler)
+: _server(SERVER_PORT)
+{
+	_chip_id = chip_id;
+	_ssid = String( String( SSID_PREFIX ) + String( _chip_id ) );
+	_ssid_prefix = String( SSID_PREFIX );
+	_handler = handler;
+}
+
+/**
+ * Wait for a WiFiClient to connect
+ *
+ * @returns: True if the client is ready, false otherwise.
+ * 
+ */
+// DEPRECATED!
+bool ESP8266WiFiMesh::waitForClient(WiFiClient curr_client, int max_wait)
+{
+	int wait = max_wait;
+	while(curr_client.connected() && !curr_client.available() && wait--)
+		delay(3);
+
+	/* Return false if the client isn't ready to communicate */
+	if (WiFi.status() == WL_DISCONNECTED || !curr_client.connected())
+		return false;
+	
+	return true;
+}
+
+/**
+ * Send the supplied message then read back the other node's response
+ * and pass that to the user-supplied handler.
+ *
+ * @target_ssid The name of the AP the other node has set up.
+ * @message The string to send to the node.
+ * @returns: True if the exchange was a succes, false otherwise.
+ * 
+ */
+// DEPRECATED!
+bool ESP8266WiFiMesh::exchangeInfo(String message, WiFiClient curr_client)
+{
+	curr_client.println( message.c_str() );
+
+	if (!waitForClient(curr_client, 1000))
+		return false;
+
+	String response = curr_client.readStringUntil('\r');
+	curr_client.readStringUntil('\n');
+
+	if (response.length() <= 2) 
+		return false;
+
+	/* Pass data to user callback */
+	_handler(response);
+	return true;
+}
+
+/**
+ * Connect to the AP at ssid, send them a message then disconnect.
+ *
+ * @target_ssid The name of the AP the other node has set up.
+ * @message The string to send to the node.
+ * 
+ */
+// DEPRECATED!
+void ESP8266WiFiMesh::connectToNode(String target_ssid, String message)
+{
+	WiFiClient curr_client;
+	WiFi.begin( target_ssid.c_str() );
+
+	int wait = 1500;
+	while((WiFi.status() == WL_DISCONNECTED) && wait--)
+		delay(3);
+
+	/* If the connection timed out */
+	if (WiFi.status() != 3)
+		return;
+
+	/* Connect to the node's server */
+	if (!curr_client.connect(SERVER_IP_ADDR, SERVER_PORT)) 
+		return;
+
+	if (!exchangeInfo(message, curr_client))
+		return;
+
+	curr_client.stop();
+	WiFi.disconnect();
+}
+
+// DEPRECATED!
+void ESP8266WiFiMesh::attemptScan(String message)
```

Comment:
Can message be a ref?

Reply:
No, that causes the example code to fail. However, we can change the private methods of CompatibilityLayer.cpp as long as we only care about keeping the public API backwards compatible. I’ve done this in the most recent commit.